### DISCUSSION TITLE
Field Comparisons

### DISCUSSION THREAD

**Discussion by critesjosh** [noir-lang/noir#3323] [Ideas]
### Problem

Being able to compare Fields in Noir would be useful. I know you can use the BigUint library for the same functionality, but I expect that uses more gates. 

### Happy Case

I can compare large Field elements without using an external library.

### Would you like to submit a PR for this Issue?

No

**Comment by Savio-Sou** [comment]
Fields currently support `==` equality checks.

Adding support for larger than / smaller than comparisons on top of that would bring us ... *drum roll* ... the unsigned integers currently in Noir 😂

**Reply by critesjosh** [reply]
The use case is to compare ethereum addresses for sorting without having to use the biguint library. Eth addresses are 160 bits and native `uint`s only go up to 127, right?

**Reply by Savio-Sou** [reply]
Hmm in that case further optimizing BigUInt's performance might be the way to go.

I recall @kevaundray / @guipublic floated around the idea of BigField that might help optimize BigUInt; any thoughts?

**Reply by guipublic** [reply]
field comparison is implemented in stdlib; `eddsa::lt_bytes32`
